### PR TITLE
[docs] Add basic parallel execution guide for Tune and cleanup order of guides

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Tune's bread and butter use case is parallel execution of trials, but we don't have any such guide for this. Add this as the first guide in the list, linking out to more advanced functionality.

### CODE REVIEW

File: doc/source/tune/doc_code/tune.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+NUM_MODELS = 100
+
+def train_model(config):
+    score = config["model_id"]
+
+    # Import model libraries, etc...
+    # Load data and train model code here...
+
+    # Return final stats. You can also return intermediate progress
+    # using Tune if needed.
```
Comment (Yard1) [inline doc/source/tune/doc_code/tune.py:18]:
```suggestion
    # using ray.air.session.report() if needed.
```

File: doc/source/tune/doc_code/tune.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    # To return your model, you could write it to storage and return its
+    # URI in this dict, or return it as a tune Checkpoint:
```
Comment (Yard1) [inline doc/source/tune/doc_code/tune.py:20]:
```suggestion
    # URI in this dict, or return it as a Tune Checkpoint:
```

File: doc/source/tune/doc_code/tune.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+trial_space = {
+    "model_id": tune.grid_search([
+        "model_{}".format(i)
+        for i in range(NUM_MODELS)
+    ])
+}
+# __step2_end__
+
+# __step3_begin__
+# Can customize resources per trial, here set 1 CPU each.
```
Comment (Yard1) [inline doc/source/tune/doc_code/tune.py:36]:
```suggestion
# Can customize resources per trial, here we set 1 CPU each.
```

**Review by Yard1 — APPROVED** [review]
Awesome! Some tiny nits

File: doc/source/tune/doc_code/tune.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    return {"score": score, "other_data": ...}
+# __step1_end__
+
+# __step2_begin__
+# Define trial parameters as a single grid sweep.
+trial_space = {
+    # This is an example parameter. You could replace it with filesystem paths,
+    # model types, or even full nested Python dicts of model configurations, etc.,
+    # that enumerate the set of trials to run.
+    "model_id": tune.grid_search([
```
Comment (xwjiang2010) [inline doc/source/tune/doc_code/tune.py:31]: nit: change this to numeric to fully mirror the other example?

**Review by xwjiang2010 — APPROVED** [review]

File: doc/source/tune/doc_code/tune.py (hunk 4 above)
Comment (ericl) [reply doc/source/tune/doc_code/tune.py:31]: I actually prefer it to be a string, to emphasize it can be an arbitrary thing and not just numbers.

**Comment by ericl** [comment]
Failures seem to be in rllib examples

**Review by richardliaw — APPROVED** [review]

**Pull request merged**